### PR TITLE
Format refactorings

### PR DESCRIPTION
**Pull request by pitdicker**
This are the 11 commits of https://github.com/chronotope/chrono/pull/1163 that are mostly just refactorings. Or at least don't add new functionality.

Does splitting them out so they can be reviewed and merge separately help?

### CODE REVIEW

File: src/format/locales.rs [hunk 1]
```diff
@@ -1,41 +1,93 @@
...
-    locale_match!(locale => LC_TIME::T_FMT)
+    pub(crate) const fn t_fmt(locale: Locale) -> &'static str {
+        locale_match!(locale => LC_TIME::T_FMT)
+    }
+
+    pub(crate) const fn t_fmt_ampm(locale: Locale) -> &'static str {
+        locale_match!(locale => LC_TIME::T_FMT_AMPM)
+    }
 }
+#[cfg(feature = "unstable-locales")]
```
Comment (djc) [inline src/format/locales.rs:45]: Style nit: please add an empty line after inline modules, here and below.

File: src/format/locales.rs [hunk 2]
```diff
@@ -1,41 +1,93 @@
...
+pub(crate) use localized::*;
+#[cfg(feature = "unstable-locales")]
+pub use pure_rust_locales::Locale;
```
Comment (djc) [inline src/format/locales.rs:48]: Maybe make this `pub use` inside the module? Seems cleaner.

**Review by djc — APPROVED** [review]
Looks great!

File: src/format/locales.rs (hunk 2 above)
Comment (pitdicker) [reply src/format/locales.rs:49]: The module is only `pub(crate)`, while this re-export should be `pub`.

Comment (djc) [reply src/format/locales.rs:49]: Have you tried it? I would expect that you can have more-public items inside a less-public module, as long as there's a more-public export.

Comment (pitdicker) [reply src/format/locales.rs:49]: This was the error:
```
error[E0365]: `Locale` is only public within the crate, and cannot be re-exported outside
  --> src\format\mod.rs:61:9
   |
61 | pub use locales::Locale;
   |         ^^^^^^^^^^^^^^^ re-export of crate public `Locale`
   |
   = note: consider declaring type or module `Locale` with `pub`

error[E0365]: `Locale` is only public within the crate, and cannot be re-exported outside
   --> src\lib.rs:432:13
    |
432 |     pub use crate::Locale;
    |             ^^^^^^^^^^^^^ re-export of crate public `Locale`
    |
    = note: consider declaring type or module `Locale` with `pub`

error[E0365]: `Locale` is only public within the crate, and cannot be re-exported outside
   --> src\lib.rs:462:9
    |
462 | pub use format::Locale;
    |         ^^^^^^^^^^^^^^ re-export of crate public `Locale`
    |
    = note: consider declaring type or module `Locale` with `pub`

For more information about this error, try `rustc --explain E0365`.
error: could not compile `chrono` (lib) due to 3 previous errors
```

**Comment by pitdicker** [comment]
Shall I merge this one first and then rebase https://github.com/chronotope/chrono/pull/1168 onto it? They are going to conflict.

File: src/format/locales.rs (hunk 2 above)
Comment (djc) [reply src/format/locales.rs:49]: Okay, fair enough!

**Comment by djc** [comment]
> Shall I merge this one first and then rebase #1168 onto it? They are going to conflict.

Whichever you prefer!

**Pull request merged**